src/fusor/models.py

```diff
@@ -77,23 +71,24 @@ class FunctionalDomain(BaseModel):
                 "type": "FunctionalDomain",
                 "status": "lost",
                 "label": "Tyrosine-protein kinase, catalytic domain",
-                "_id": "interpro:IPR020635",
-                "associated_gene": {
+                "id": "interpro:IPR020635",
+                "gene": {
                     "id": "gene:NTRK1",
                     "gene_id": "hgnc:8031",
                     "label": "8031",
-                    "type": "GeneDescriptor",
+                    "type": "Gene",
                 },
-                "sequence_location": {
-                    "id": "fusor.location_descriptor:NP_002520.2",
-                    "type": "LocationDescriptor",
-                    "location": {
-                        "sequence_id": "ga4gh:SQ.vJvm06Wl5J7DXHynR9ksW7IK3_3jlFK6",
-                        "type": "SequenceLocation",
-                        "interval": {
-                            "start": {"type": "Number", "value": 510},
-                            "end": {"type": "Number", "value": 781},
-                        },
+                "sequenceLocation": {
+                    "id": "SL.vJvm06Wl5J7DXHynR9ksW7IK3_3jlFK6",
+                    "start": 510,
+                    "end": "781",
+                    "type": "SequenceLocation",
+                    "sequenceReference": {
+                        "id": "NP_002520.2",
+                        "label": "GRCh38:chr22",
+                        "type": "SequenceReference",
+                        "refgetAccession": "SQ.7B7SHsmchAR0dFcDCuSFjJAo7tX87krQ",
+                        "residueAlphabet": "na",
```

Comment:
I think this is a little confusing. `NP_002520.2` is on the protein reference sequence, but `GRCh38:chr22` and `SQ.7B7SHsmchAR0dFcDCuSFjJAo7tX87krQ` indicate that this is on the genomic reference sequence. So I think id should reflect the genomic reference sequence.
```suggestion
                        "id": "NC_000022.11",
                        "label": "GRCh38:chr22",
                        "type": "SequenceReference",
                        "refgetAccession": "SQ.7B7SHsmchAR0dFcDCuSFjJAo7tX87krQ",
                        "residueAlphabet": "na",
```